persistence-api/src/main/java/io/stargate/db/util/TimeSource.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.db.util;
```

Comment:
For SGv2, we're trying to eliminate dependence of the API Services on the `persistence-api`. Since the document API implementation uses this class, and you're now proposing to use it in the persistence implementations, it might be better to move this interface to the`core` module.

Reply:
Hmm, there was a reason it's not in the `core`.. I will double check, on the first look seems possible yea..

Reply:
Moved to `core`, let's see if tests are green..